csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs

```diff
@@ -91,22 +93,27 @@ public DependencyManager(string srcDir, IDependencyOptions options, ILogger logg
                     progressMonitor.MissingNuGet();
                 }
 
-                var restoredProjects = RestoreSolutions(solutions);
+                var restoredProjects = RestoreSolutions(solutions, out var assets1);
                 var projects = allProjects.Except(restoredProjects);
-                RestoreProjects(projects);
+                RestoreProjects(projects, out var assets2);
+
+                var dependencies = Assets.GetCompilationDependencies(progressMonitor, assets1.Union(assets2));
+                existsNetCoreRefNugetPackage = dependencies.Any(dependency => dependency.StartsWith("microsoft.netcore.app.ref"));
+                existsNetFrameworkRefNugetPackage = dependencies.Any(dependency => dependency.StartsWith("microsoft.netframework.referenceassemblies"));
+
+                var paths = dependencies
+                    .Select(d => Path.Combine(packageDirectory.DirInfo.FullName, d))
+                    .ToList();
+
+                // TODO: Rename the dllDirNames var - it's not only dirs anymore.
+                dllDirNames.AddRange(paths);
                 DownloadMissingPackages(allNonBinaryFiles);
```

Comment:
We don't add `packageDirectory.DirInfo.FullName` any longer to `dllDirNames`. Doesn't this mean that `DownloadMissingPackages` might fetch packages, but those are never used by the extraction?

Reply:
Yes, this looks wrong!